Server/Server.Test/ServerConnectionTests.cpp

```diff
@@ -38,13 +42,351 @@ TEST_CASE("Server Test")
+    SafeQueue<Message> testIncomingMessagesQueue;
+
+    auto connectionPointer = std::make_shared<Connection>(Connection::OwnerType::SERVER, testContext, asio::ip::tcp::socket(testContext),
+                                                          testIncomingMessagesQueue);
+
+    uint16_t port;
+
+    REQUIRE(port <= 65003);
+    Server::Server server(port);
+
+    server.start();
+
+    const std::string email    = "demonstakingoverme@epam.co";
+    const std::string login    = "devilishTrio_g12";
+    const std::string password = "2003stasStanislavov";
+    const std::string pwdHash  = Base::Hashing::SHA_256(password, login);
+
+    Network::RegistrationInfo registrationInfo(email, login, pwdHash);
+
+    SECTION("Check registration request of server") 
+    {
+        Message registrationRequest;
+
+        registrationRequest.mHeader.mMessageType = Message::MessageType::RegistrationRequest;
+        registrationRequest.mHeader.mTimestamp   = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+        registrationRequest.mBody = std::make_any<Network::RegistrationInfo>(registrationInfo);
+
+        REQUIRE_NOTHROW(server.registrationRequest(connectionPointer, registrationRequest));
+    }
+
+    SECTION("Check login request of server") 
+    {
+        Network::LoginInfo loginInfo(login, pwdHash);
+        
+        Message loginRequest;
+        loginRequest.mHeader.mMessageType = Message::MessageType::LoginRequest;
+        loginRequest.mBody                = std::make_any<Network::LoginInfo>(loginInfo);
+
+        REQUIRE_NOTHROW(server.loginRequest(connectionPointer, loginRequest));
+    }
+
+    server.stop();
+}
+
+TEST_CASE("Testing methods of server reading & storing messages") 
+{
+    asio::io_context   testContext;
+    SafeQueue<Message> testIncomingMessagesQueue;
+
+    auto connectionPointer = std::make_shared<Connection>(Connection::OwnerType::SERVER, testContext, asio::ip::tcp::socket(testContext),
+                                                          testIncomingMessagesQueue);
+
+    uint16_t port;
+
+    REQUIRE(port <= 65003);
+    Server::Server server(port);
+
+    server.start();
+
+    SECTION("Check reading of all messages of server") 
+    {
+        //TODO Add messages in queue
+        Message allMessages;
+        allMessages.mHeader.mMessageType = Message::MessageType::MessageAll;
+        allMessages.mHeader.mTimestamp   = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+        
+        REQUIRE_NOTHROW(server.readAllMessage(connectionPointer, allMessages));
+    }
+
+    SECTION("Check messages history request of server") 
+    {
+        //TODO Adding messages
+        const std::uint64_t channelID = 0;
+
+        Network::Message historyRequest;
+        historyRequest.mHeader.mMessageType = Message::MessageType::MessageHistoryRequest;
+        historyRequest.mHeader.mTimestamp   = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
```

Comment:
In most of the cases, you could encapsulate message creation in function. This decrease the count of code